src/resolvers/archDockerfile.ts

```diff
@@ -25,11 +40,15 @@ export class ArchDockerfileResolver implements Resolver {
 		this.archDockerfiles.push([ext, file]);
 	}
 
-	public needsEntry(filepath: string): boolean {
-		const filename = path.basename(filepath);
-		return (
-			filename.startsWith('Dockerfile.') && !filename.endsWith('.template')
-		);
+	public needsEntry(
+		entry: TarEntryPath,
+		specifiedDockerfilePath?: string,
+	): boolean {
+		const nameMatches = specifiedDockerfilePath
+			? entry.path === specifiedDockerfilePath
+			: entry.pathNoExt === 'Dockerfile';
+
+		return nameMatches && !!entry.extension && entry.extension !== '.template';
```

Comment:
Two cases:
* If a specifiedDockeriflePath was specified, say `'service/MyDockerfile.armv7hf'`, then this resolver (`ArchDockerfileResolver`) will only match a tar entry whose full path is identical to that (`entry.path === specifiedDockerfilePath`), and provided that the specifiedDockeriflePath has a non-empty file extension different to 'template'.
* Otherwise, it will match `Dockerfile.xxx` where xxx is a non-empty file extension different to 'template' (as per previous logic). This will only be matched at the root of the project directory tree, as `pathNoExt` is the full path (minus file extension) which must be exactly `'Dockerfile'`. This root-only rule was previously enforced in `resolveTarStreamOnEntry()` through a test: `if (path.normalize(name).split(path.sep).length === 1)`. That test had been added by a recent-ish pull request:
_fix: Correctly resolve contexts with multiple bundle descriptions_
https://github.com/balena-io-modules/resin-bundle-resolve/pull/32/files